src/app/app.ts

```diff
@@ -20,50 +20,10 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
-import { EditorState, EditorView } from '@codemirror/next/basic-setup';
+import { basicSetup, EditorState, EditorView } from '@codemirror/next/basic-setup';
 import { PromQLExtension } from '../lang-promql';
-import { Extension } from '@codemirror/next/state';
-import { history, historyKeymap } from '@codemirror/next/history';
-import { highlightSpecialChars, keymap, multipleSelections } from '@codemirror/next/view';
-import { lineNumbers } from '@codemirror/next/gutter';
-import { foldGutter, foldKeymap } from '@codemirror/next/fold';
-import { bracketMatching } from '@codemirror/next/matchbrackets';
-import { closeBrackets, closeBracketsKeymap } from '@codemirror/next/closebrackets';
-import { autocomplete, autocompleteKeymap } from '@nexucis/codemirror-next-autocomplete';
-import { rectangularSelection } from '@codemirror/next/rectangular-selection';
-import { highlightActiveLine, highlightSelectionMatches } from '@codemirror/next/highlight-selection';
-import { defaultKeymap } from '@codemirror/next/commands';
-import { searchKeymap } from '@codemirror/next/search';
-import { commentKeymap } from '@codemirror/next/comment';
-import { gotoLineKeymap } from '@codemirror/next/goto-line';
-import { lintKeymap } from '@codemirror/next/lint';
 import { promQLHighlightMaterialTheme } from './theme';
 
-const basicSetup: Extension = [
-  lineNumbers(),
-  highlightSpecialChars(),
-  history(),
-  foldGutter(),
-  multipleSelections(),
-  bracketMatching(),
-  closeBrackets(),
-  autocomplete({ matchPre: '<b style="color: brown">', matchPost: '</b>' }),
-  rectangularSelection(),
-  highlightActiveLine(),
-  highlightSelectionMatches(),
-  keymap([
-    ...closeBracketsKeymap,
-    ...defaultKeymap,
-    ...searchKeymap,
-    ...historyKeymap,
-    ...foldKeymap,
-    ...commentKeymap,
-    ...gotoLineKeymap,
-    ...autocompleteKeymap,
-    ...lintKeymap,
-  ]),
-];
-
```

Comment:
Just want to say that making the example simpler is great! Thank you

Reply:
:D the fork was just there to speed up the development and the integration of this lib into Prometheus and PromLens.
It was also a good way to test if cmn is working properly with another implementation of the autocompletion module.

But even if CMN is working as a module, it is still simpler to use the vanilla modules. It simplify so many things !

Reply:
and you are really welcome :)